### PR TITLE
BIOIN-1755-add_single_cell

### PR DESCRIPTION
**Pull request by inbalzelig-ugi**
1. Add new pacakge for single cell.
2. Add utils to core.
3. Add licenses (for ugvc and star).
4. Update CI and build action

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -1 +0,0 @@
-3.10.14
```
Comment (avigailmo) [inline .python-version:1]: I'm not sure it's safe to delete this file. It was needed for something (I don't remember exactly what)

File: src/single_cell/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "matplotlib>=3.9.1",
+    "numpy<2.0.0", # New versions of numpy are uncompatible, not sure why pandas doesn't have this constrain. See https://github.com/numpy/numpy/issues/26710 to get a hunch of the issue
+    "ipykernel>=6.29.5",
+    
+]
+description = "Ultima Genomics Single Cell utils"
+authors = [{ name = "Inbal Zelig", email = "inbal.zelig@ultimagen.com" }]
+readme = "README.single_cell.md"
+
+[project.scripts]
```
Comment (avigailmo) [inline src/single_cell/pyproject.toml:22]: 👍

File: licenses/star LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (avigailmo) [inline licenses/star LICENSE.txt:1]: those licenses files don't need to be referenced somehow from the pyproject.toml file?

**Review by avigailmo — COMMENTED** [review]
Generally looks very good and clean!
My main concern is about deleting the .python-version file- I don't remember what but I do remember I had to add it for some reason
Please make sure to update ugbio-utils submodule in VariantCalling and verify ci is working - before merging

File: src/core/pyproject.toml [hunk 4]
```diff
@@ -1,13 +1,14 @@
 [project]
 name = "ugbio_core"
 version = "0.1.0.dev9"
-requires-python = ">= 3.8"
+requires-python = ">= 3.10"
```
Comment (avigailmo) [inline src/core/pyproject.toml:4]: 👍

File: src/core/pyproject.toml [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
 description = "Ultima Genomics core bioinfo utils"
 authors = [{ name = "Avigail Moldovam ", email = "avigail.moldovan@ultimagen.com" }]
 readme = "README.core.md"
 license = { text = "Apache-2.0" }
 dependencies = [
-    "simppl==1.0.7"
+    "simppl==1.0.7",
+    "pandas>=2.2.2",
```
Comment (avigailmo) [inline src/core/pyproject.toml:11]: have you verified this pandas upgrade with Ilya?

File: src/single_cell/tests/resources/trimmer_stats.csv.gz
Comment (avigailmo) [inline src/single_cell/tests/resources/trimmer_stats.csv.gz:1]: this doesn't look like gz file

**Review by avigailmo — COMMENTED** [review]
generally looks good. added two comments
Can you please refer me to the pr in terra_pipelines that uses this single_cell docker?

File: src/single_cell/tests/resources/trimmer_stats.csv.gz
Comment (inbalzelig-ugi) [reply src/single_cell/tests/resources/trimmer_stats.csv.gz:1]: You are right. File removed

**Review by avigailmo — APPROVED** [review]

**Pull request merged**